CRM/Core/OptionGroup.php

```diff
@@ -611,4 +599,30 @@ public static function flushAll() {
     CRM_Utils_Cache::singleton()->flush();
   }
 
+  /**
+   * Uses api adapter to fetch `from_email_address` options as if it were still an option group
+   *
+   * @see DomainEmailLegacyOptionValueAdapter
+   * @throws CRM_Core_Exception
+   */
+  private static function getLegacyFromEmailAddressValues(bool $onlyActive, $condition, string $labelColumnName, string $keyColumnName): array {
+    $where = [
+      ['option_group_id:name', '=', 'from_email_address'],
+      ['domain_id', '=', 'current_domain'],
+    ];
+    if ($onlyActive) {
+      $where[] = ['is_active', '=', TRUE];
+    }
+    if ($condition && is_string($condition)) {
+      // Use regex to extract the field_name and the value from the condition.
+      if (preg_match("/([\w_]+)\s*=\s*(\d+)$/", $condition, $matches)) {
```

Comment:
Currently, the condition is ignored if it does not match the regex. Is that ok, or should it throw an error if the condition is an unexpected format?

Reply:
Thanks for the suggestion. I agree & have updated the PR.